sympy/integrals/risch.py

```diff
@@ -1109,7 +1109,7 @@ def hermite_reduce(a, d, DE):
         ga, gd = ga.cancel(gd, include=True)
         dm = dm2
 
-    d = ds
+        d = ds
```

Comment:
Was this intentional? neither `d` nor `ds` is updated in the loop. A better change might be to change 1090 to `d, r = d.div(dm)` and change `ds` in the loop to `d`.  But...again, this didn't fix anything, did it? Perhaps it just confirms that the issue is fixed (or else is intermittant).

Reply:
if `dm.degree(DE.t)` is 0 to start with you get a different value for `d`. So, yes, it actually did change the behaviour.

Reply:
But you are correct that one can write it in more clear way. I'll update.